l2-contracts/contracts/interfaces/IConsensusRegistry.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+import {Ownable2Step} from "@openzeppelin/contracts/access/Ownable2Step.sol";
+import {EfficientCall} from "@matterlabs/zksync-contracts/l2/system-contracts/libraries/EfficientCall.sol";
+
+/// @author Matter Labs
+/// @custom:security-contact security@matterlabs.dev
+/// @title ConsensusRegistry contract interface
+interface IConsensusRegistry {
+    /// @dev Represents a consensus node.
+    /// @param attesterLastUpdateCommit The latest `attestersCommit` where the node's attester attributes were updated.
+    /// @param attesterLatest Attester attributes to read if `node.attesterLastUpdateCommit` < `attestersCommit`.
+    /// @param attesterSnapshot Attester attributes to read if `node.attesterLastUpdateCommit` == `attestersCommit`.
+    /// @param validatorLastUpdateCommit The latest `validatorsCommit` where the node's validator attributes were updated.
+    /// @param validatorLatest Validator attributes to read if `node.validatorLastUpdateCommit` < `validatorsCommit`.
+    /// @param validatorSnapshot Validator attributes to read if `node.validatorLastUpdateCommit` == `validatorsCommit`.
+    struct Node {
+        uint256 attesterLastUpdateCommit;
+        AttesterAttr attesterLatest;
+        AttesterAttr attesterSnapshot;
+        uint256 validatorLastUpdateCommit;
+        ValidatorAttr validatorLatest;
+        ValidatorAttr validatorSnapshot;
+    }
+
+    /// @dev Represents the attester attributes of a consensus node.
+    /// @param active A flag stating if the attester is active.
+    /// @param pendingRemoval A flag stating if the attester is pending removal.
+    /// @param weight Attester's voting weight.
+    /// @param pubKey Attester's Secp256k1 public key.
+    struct AttesterAttr {
+        bool active;
+        bool pendingRemoval;
```

Comment:
I had the same thought but the commit numbers increase independently, so I thought that means you can have a node that is still an active validator while it’s a removed attester at the same time.

Reply:
As @aakoshh mentioned, since committees are committed to independently, data duplication is needed. And from that perspective, the concept of a node doesn't really exist, but rather distinct validator and attester roles.